### PR TITLE
feat: remove empty folders on python uninstall

### PR DESCRIPTION
**Pull request by baszalmstra**
Remove empty folders when uninstalling a pypi dependency.

### CODE REVIEW

File: src/environment.rs [hunk 1]
```diff
@@ -534,14 +535,15 @@ fn remove_old_python_distributions(
...
             .expect("the dist-info path must be a sub-path of the site-packages path");
-        rip::uninstall::uninstall_distribution(&prefix.root().join(site_package_path), relative_dist_info)
-            .into_diagnostic()
-            .with_context(|| format!("could not uninstall python package {}-{}. Manually remove the `.pixi/env` folder and try again.", &python_package.name, &python_package.version))?;
 
         // HACK: Also remove the HASH file that pixi writes. Ignore the error if its there. We
         // should probably actually add this file to the RECORD.
         let _ = std::fs::remove_file(prefix.root().join(&python_package.dist_info).join("HASH"));
 
+        rip::uninstall::uninstall_distribution(&prefix.root().join(site_package_path), relative_dist_info)
```
Comment (ruben-arts) [inline src/environment.rs:543]: This does look like some code duplication, can't uninstalling be moved into a function?

Comment (baszalmstra) [reply src/environment.rs]: Yes! Done :)

**Comment by ruben-arts** [comment]
```bash
pixi/examples/pypi on  fix/remove_uninstalled_python_dist [!?] 
❯ git diff pixi.toml
diff --git a/examples/pypi/pixi.toml b/examples/pypi/pixi.toml
index 06fb846..5eae860 100644
--- a/examples/pypi/pixi.toml
+++ b/examples/pypi/pixi.toml
@@ -17,7 +17,8 @@ libclang = "~=16.0.6"
 pyboy = "==1.6.6"
 tensorflow = "==2.14.0"
 flask = "*"
-black = "~=23.10"
+#black = "~=23.10"
+
 
 [system-requirements]
 # Tensorflow on macOS arm64 requires macOS 12.0 or higher

pixi/examples/pypi on  fix/remove_uninstalled_python_dist [!?] 
➜ tree .pixi/env/lib/python3.11/site-packages/black*
.pixi/env/lib/python3.11/site-packages/black
.pixi/env/lib/python3.11/site-packages/black-23.11.0.dist-info
└── licenses
.pixi/env/lib/python3.11/site-packages/blackd
.pixi/env/lib/python3.11/site-packages/black.libs

2 directories, 0 files

pixi/examples/pypi on  fix/remove_uninstalled_python_dist [!?] 
```
Not sure why this doesn't work for me.

**Review by ruben-arts — APPROVED** [review]
Never mind this worked, looks good to me!

**Pull request merged**